sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -260,6 +262,56 @@ def to_runner_api(self, unused_context):
         default=beam_runner_api_pb2.Trigger.Default())
 
 
+class AfterProcessingTime(TriggerFn):
+  """Fire exactly once after a specified delay from processing time.
+
+  AfterProcessingTime is experimental. No backwards compatibility guarantees.
+  """
+
+  def __init__(self, delay=0):
+    self.delay = delay
+
+  def __repr__(self):
+    return 'AfterProcessingTime(delay=%d)' % self.delay
+
+  def on_element(self, element, window, context):
+    context.set_timer(
+        '', TimeDomain.REAL_TIME, context.get_current_time() + self.delay)
+
+  def on_merge(self, to_be_merged, merge_result, context):
+    for window in to_be_merged:
+      if window.end != merge_result.end:
+        context.clear_timer('', TimeDomain.REAL_TIME)
+
+  def should_fire(self, watermark, window, context):
+    # evaluation deferred to be handled by the
+    # GeneralTriggerDriver's process_timer routine
```

Comment:
<!--thread_id:cc_163459889_t; commit:a05ebd729a88b55bbf37a2e0c501299a923aab08; resolved:0-->
<!--section:context-quote-->
> **charlesccychen** wrote:
> Please fix capitalization.
> 
> Can you please add a TODO indicating that once `TriggerFn.should_fire` receives watermark domains, to 
> 
> (context for Robert: this has to be False for now since `should_fire` currently only receives watermark time domain timestamps)

<!--section:body-->
Your message appears cut, so I am not clear on what the request it. I did add a TODO in the GeneralTriggerDriver to move the code to handle realtime timers to here (if that is what you mean).